docs/pages/wallet-security/secure-multisig-best-practices.mdx

```diff
@@ -144,6 +153,14 @@ related to the multisig, including proposed transactions, new signatures, and ow
 * **Documented Procedures:** Maintain clear, secure, and accessible documentation for all multisig procedures, including transaction creation, signing, and emergency recovery plans.
 * **General Signing Guidelines:** Use hardware wallets, dedicated signing profiles, and re-verify before execution. Require a "how to check" guide and communicate status after signing (e.g., "checked, signed, X more required"). Last signer executes when applicable.
 
+## Contract-Level Security
+
+* **Invariant Enforcement:** Design contracts to enforce invariants and expected state changes such as token balance changes, ownership and administration, and proxy implementation addresses. Ensure contracts automatically revert transactions if any invariant is violated.
```

Comment:
I'm a bit confused as to how invariant enforcement ties in to multisigs. This seems like it belongs in a section on smart contract development.

Reply:
If we want to cover invariants for multisigs, we should probably link out to post state checking baked into transactions which is something I could have sworn @engn33r mentioned. Otherwise, I would think this falls under monitoring multisig tx's